src/new/templates/rust/no-ui/fibonacci/{package_name}/src/lib.rs

```diff
@@ -18,96 +17,97 @@ enum FibonacciRequest {
-                        let _result = fibonacci(number);
-                        let duration = start.elapsed();
-                        durations.push(duration);
-                    }
-                    let result = fibonacci(number);
-                    let mean = durations
-                        .iter()
-                        .fold(0, |sum, item| sum + item.as_nanos()) / number_trials as u128;
-                    let absolute_deviation = durations
-                        .iter()
-                        .fold(0, |ad, item| {
-                            let trial = item.as_nanos();
-                            ad + if mean >= trial { mean - trial } else { trial - mean }
-                        }) / number_trials as u128;
-                    println!(
-                        "{package_name}: fibonacci({}) = {}; {}±{}ns averaged over {} trials",
-                        number,
-                        result,
-                        mean,
-                        absolute_deviation,
-                        number_trials,
-                    );
-                    Response::new()
-                        .body(serde_json::to_vec(&FibonacciResponse::Numbers((
-                            result,
-                            number_trials,
-                        ))).unwrap())
-                        .send()
-                        .unwrap();
-                },
+    if !message.is_request() {
+        return Err(anyhow::anyhow!("expected a request"));
+    }
+
+    match serde_json::from_slice(message.body())? {
+        FibonacciRequest::Number(number) => {
+            let start = std::time::Instant::now();
+            let result = fibonacci(number);
+            let duration = start.elapsed();
+            println!(
+                "{package_name}: fibonacci({}) = {}; {}ns",
+                number,
+                result,
+                duration.as_nanos(),
+            );
+            Response::new()
+                .body(serde_json::to_vec(&FibonacciResponse::Number(result)).unwrap())
+                .send()
+                .unwrap();
+        }
+        FibonacciRequest::Numbers((number, number_trials)) => {
+            let mut durations = Vec::new();
+            for _ in 0..number_trials {
+                let start = std::time::Instant::now();
+                let _result = fibonacci(number);
+                let duration = start.elapsed();
+                durations.push(duration);
             }
-        },
+            let result = fibonacci(number);
+            let mean =
+                durations.iter().fold(0, |sum, item| sum + item.as_nanos()) / number_trials as u128;
+            let absolute_deviation = durations.iter().fold(0, |ad, item| {
+                let trial = item.as_nanos();
+                ad + if mean >= trial {
+                    mean - trial
+                } else {
+                    trial - mean
+                }
+            }) / number_trials as u128;
```

Comment:
nit: The above was much more readable to me as the one-liner
```
ad + if mean >= trial { mean - trial } else { trial - mean }
```

Reply:
rustfmt automatically puts it on 4 lines :(